### ISSUE TITLE
Clicking the "generate message" button causes the IDEA commit panel to freeze directly.

### ISSUE THREAD

**Issue by kenzson** [carlrobertoh/CodeGPT#348]
### What happened?

Clicking the "generate message" button causes the IDEA commit panel to freeze directly. However, the chat function can be used normally. Currently, IDEA is using version 2023.3.3, and CodeGPT is using version 2.2.12.
![Screenshot 2024-02-02 160502](https://github.com/carlrobertoh/CodeGPT/assets/11890025/c6c7698a-2d1a-435b-b280-f1122baa8f34)

### CodeGPT version

2.2.12

### Operating System

Windows

**Comment by kenzson** [comment]
This seems to be a problem caused by CodeGPT using `git diff`. Can SVN projects not use the "generate message" feature? What should I do?

**Comment by carlrobertoh** [comment]
I'm unable to reproduce this. Is this still happening?

**Comment by kenzson** [comment]
Yes, the issue still exists even after updating to the latest version. My project uses SVN version control. Clicking the "Generate Message" button in the commit panel still causes the IDE to freeze. I need to find the Git process in my task manager and end it to restore the IDE from freezing.

**Comment by kenzson** [comment]
![image](https://github.com/carlrobertoh/CodeGPT/assets/11890025/011ed098-2d10-4b14-b678-8a570cd9f9a4)
![image](https://github.com/carlrobertoh/CodeGPT/assets/11890025/a6e91489-e336-4abb-9a4e-f2c9df3fce07)
My computer only has one disk, which is the C drive, the system disk.

**Comment by kenzson** [comment]
I found that CodeGPT only supports automatically generating commit messages for Git projects. Could you please add support for projects using Subversion version control? @carlrobertoh

**Comment by BenisonSam** [comment]
For me the whole IDE freezes and every time I use this feature, I have to kill the IDE and start it again.
I am using Git.
